plugins/inputs/t128_tank/reader.go

```diff
@@ -154,10 +154,62 @@ func (r *Reader) Run(mainCtx context.Context) {
 				r.lastSavedIndex = lastIndex.value
 			}
 			nextSaveCheck = time.After(2 * time.Second)
+		case err := <-r.readDone:
+			var errBoundaryFault *boundaryFault
+			if err != nil && errors.As(err, &errBoundaryFault) {
+				r.log.Debugf("detected boundary fault, restarting %s tank read from index %d", r.topic, errBoundaryFault.nextAvailableIndex.value)
+				go r.read(mainCtx, readCtx, errBoundaryFault.nextAvailableIndex)
+			} else {
+				go r.read(mainCtx, readCtx, lastIndex.next())
+			}
 		}
 	}
 }
 
+func (r *Reader) getIndex(indexPath string, defaultIndex index) (index, error) {
+	if indexPath == "" {
+		r.log.Debugf("index file path not provided, starting with default index %s", defaultIndex.string())
+		return defaultIndex, nil
+	}
+	content, err := os.ReadFile(indexPath)
+	if errors.Is(err, os.ErrNotExist) {
+		r.log.Debugf("index file %s does not exist, starting with default index %s", indexPath, defaultIndex.string())
+		return defaultIndex, nil
+
+	} else if err != nil {
+		return defaultIndex, fmt.Errorf("encountered error reading index file, starting with default index %s: %s", defaultIndex.string(), err)
+	}
+
+	r.log.Debugf("found '%s' in index file", content)
+
+	index, err := newIndex(string(content))
+	if err != nil {
+		return defaultIndex, fmt.Errorf("encountered error while parsing index file content, starting with default index %s: %s", defaultIndex.string(), err)
+	}
+
+	return index, nil
+}
+
+func (r *Reader) setIndex(indexPath string, index index) {
+	if indexPath == "" {
+		r.log.Debugf("index file path not provided, starting with index %d", index.value)
+		return
+	}
+	_, err := os.Stat(path.Dir(indexPath))
+	if errors.Is(err, os.ErrNotExist) {
+		err := os.MkdirAll(path.Dir(indexPath), os.ModePerm)
+		if err != nil {
+			r.log.Debugf("unable to create index file parent directory: %s", err)
+			return
+		}
+	}
+
+	err = os.WriteFile(indexPath, []byte(index.string()), 0644)
```

Comment:
use `0600`

Reply:
want to keep it accessible by owner I am assuming?